server/src/test/java/io/crate/analyze/UnionAnalyzerTest.java

```diff
@@ -197,9 +197,29 @@ public void testUnionObjectTypesWithSubColumnsOfSameNameButDifferentInnerTypes()
             ).build();
 
         UnionSelect union = analyze("select obj from v1 union all select obj from v2");
-        ObjectType expectedType = ObjectType.builder().setInnerType("col", DataTypes.STRING).build();
+        ObjectType expectedType = ObjectType.builder().setInnerType("col", DataTypes.INTEGER).build();
         Asserts.assertThat(union.outputs())
-            .as("Output type is object(col::text) because text has higher precedence than int")
+            .as("Output type is object(col::int) because int has higher precedence than text")
+            .satisfiesExactly(isField("obj", expectedType));
+    }
+
+    @Test
+    public void test_union_object_types_can_have_sub_columns_from_either_sides_of_objects() throws Exception {
+        SQLExecutor.builder(clusterService)
+            .addTable(
+                "create table v1 (obj object as (col1 int, col2 text))"
```

Comment:
same as in docs, maybe add a `col3` to show the merging of data types at the analysis level?
(not insisting since we have the data types unit testing)